File: jmlparser-jml-tools/src/main/java/com/github/jmlparser/redux/ReduxConfig.java
package com.github.jmlparser.redux;

import java.util.HashSet;
import java.util.Set;

/**
 * Configuration for the Redux transformation pipeline.
 */
public class ReduxConfig {
    private Set<String> disabled = new HashSet<>();

    public void enable(Class<?> clazz) {
        disabled.remove(clazz.toString());
    }

    public void disable(Class<?> clazz) {
        disabled.add(clazz.toString());
    }

    public Set<String> getDisabled() {
        return disabled;
    }

    public boolean isEnabled(String clazz) {
        return !disabled.contains(clazz);
    }
}

File: tools/redux/src/main/kotlin/io/github/jmltoolkit/redux/ReduxConfig.kt
package io.github.jmltoolkit.redux


/**
 * Configuration for the Redux transformation pipeline.
 */
class ReduxConfig {
    private val disabled: MutableSet<String> = HashSet()

    fun enable(clazz: Class<*>) {
        disabled.remove(clazz.toString())
    }

    fun disable(clazz: Class<*>) {
        disabled.add(clazz.toString())
    }

    fun getDisabled(): Set<String> {
        return disabled
    }

    fun isEnabled(clazz: String): Boolean {
        return !disabled.contains(clazz)
    }
}